### PR TITLE
Add provisioning post

### PR DESCRIPTION
**Pull request by fhunleth**

### CODE REVIEW

File: content/post/2018-06-15-serial_number.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+way.
+
+The first step is to decide on a way of identifying devices. Nearly every
+Nerves System has a way of finding a unique identifier for a board by default.
+The default is to use that number to create a unique hostname and you can see
+it by running `:inet.gethostname()`. It will be something like `nerves-1234`.
+(This hostname is only local unless something registers it in the DNS.
+`nerves.local` is a mDNS name.) The `erlinit.config` file specifies how to
+create the hostname. For the [Raspberry Pi 3](rpi3_erlinit) look for the
+following:
```
Comment (mobileoverlord) [inline content/post/2018-06-15-serial_number.md]: > For the [Raspberry Pi 3](rpi3_erlinit) look for the following:

I am thinking

In the Raspberry Pi 3 [erlinit.config](rpi3_erlinit) look for the following lines:

File: content/post/2018-06-15-serial_number.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+```elixir
+-d "/usr/bin/boardid -b uboot_env -u serial_number -b rpi -n 4"
+-n nerves-%s
+```
+
+The `-d` setting specifies how to find a unique ID. This invokes
+[boardid](boardid) to look it up. Don't worry about the commandline arguments
+yet. The unique ID could be stored in the CPU (like on the Raspberry Pi), an
+EEPROM, a few other places depending on the board. The `-n` setting specifies
```
Comment (mobileoverlord) [inline content/post/2018-06-15-serial_number.md]: > EEPROM, a few other places depending on the board.

EEPROM, or a few other places depending on the board.

File: content/post/2018-06-15-serial_number.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+the format of the hostname where the `%s` is substituted for the identifier.
+
+Getting back to the `boardid` commandline, the arguments say to use the
+`serial_number` key from the U-Boot environment first and if that doesn't exist,
```
Comment (mobileoverlord) [inline content/post/2018-06-15-serial_number.md]: > key from the U-Boot environment first and if that doesn't exist,

Should this be an or?
key from the U-Boot environment first or if that doesn't exist,

File: content/post/2018-06-15-serial_number.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+stored outside of any of the filesystems.
+[Nerves.Runtime](nerves_runtime_metadata) documents the keys that Nerves uses. A
+device doesn't have to run the U-Boot bootloader to have a U-Boot environment.
+The format is convenient so Nerves reuses it.
+
+Nerves does not make writing values to the U-Boot environment convenient to
+reduce the chance of corrupting or losing data in it. Users should prefer to
+store application settings and data in the Nerves application partition. The
+U-Boot environment is appropriate for provisioning information that is unlikely
+to change over the device's lifetime, though. The serial number of the device is
```
Comment (mobileoverlord) [inline content/post/2018-06-15-serial_number.md]: > to change over the device's lifetime, though.

I don't think we need the ", though" here

**Comment by fhunleth** [comment]
@mobileoverlord Thanks for the comments - I addressed all of them.

**Pull request merged**